SConstruct

```diff
@@ -184,8 +184,12 @@ for k in platform_opts.keys():
 modules_detected = {}
 module_search_paths = ["modules"]  # Built-in path.
 
-if ARGUMENTS.get("custom_modules"):
-    paths = ARGUMENTS.get("custom_modules").split(",")
+# Update the environment now as the "custom_modules" option may be
+# defined in a file rather than specified via the command line.
+opts.Update(env_base)
```

Comment:
I'd move this block just after the last `opts.Add(o)` above (platform-specific options), as it makes sense to update the env once all options have been registered.